pkg/controller/node/nodecontroller.go

```diff
@@ -334,13 +375,20 @@ func NewNodeController(
 						return
 					}
 				}
+				if nc.taintManager != nil {
+					nc.taintManager.NodeUpdated(node, nil)
+				}
 				if err := nc.cidrAllocator.ReleaseCIDR(node); err != nil {
 					glog.Errorf("Error releasing CIDR: %v", err)
 				}
 			},
 		}
 	}
 
+	if nc.runTaintManager {
+		nc.taintManager = NewNoExecuteTaintManager(kubeClient)
```

Comment:
This looks a lot like a discrete controller.  Is there some hard link between these two that isn't observable via the API and forces these to be linked in the same controller?

If this were separate, it would be able to use our "normal" enable/disable flags and might end up more similar to other controllers for maintenance.